### PR TITLE
[frameworks] Add type annotations

### PR DESCRIPTION
**Pull request by TooTallNate**
This will be necessary for static type analysis in our API docs generator.

### CODE REVIEW

File: packages/frameworks/src/types.ts [hunk 1]
```diff
@@ -1,50 +1,174 @@
...
+   */
   slug: string | null;
+  /**
+   * A URL to the logo of the framework
+   * @example https://raw.githubusercontent.com/vercel/vercel/master/packages/frameworks/logos/next.svg
+   */
   logo: string;
+  /**
+   * A URL to a deployed example of the framework
+   * @example https://nextjs.now-examples.now.sh
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
   * @example https://nextjs.now-examples.vercel.app
```

File: packages/frameworks/src/types.ts [hunk 2]
```diff
@@ -1,50 +1,174 @@
...
+   */
+  useRuntime?: {
+    /**
+     * Runtime source
+     * @example "package.json"
+     */
+    src: string;
+    /**
+     * Runtime
+     * @example @vercel/next
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
     * @example "@vercel/next"
```

File: packages/frameworks/src/types.ts [hunk 3]
```diff
@@ -1,50 +1,174 @@
...
     every?: FrameworkDetectionItem[];
+    /**
+     * Collection of detectors where one match triggers the framework
+     * to be detected.
+     */
     some?: FrameworkDetectionItem[];
   };
   settings: {
+    /**
+     * Default install command or a placeholder
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
     * Default Install Command or a placeholder
```

File: packages/frameworks/src/types.ts [hunk 4]
```diff
@@ -1,50 +1,174 @@
...
+     */
     installCommand: Setting;
+    /**
+     * Default build command or a placeholder
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
     * Default Build Command or a placeholder
```

File: packages/frameworks/src/types.ts [hunk 5]
```diff
@@ -1,50 +1,174 @@
...
+     */
     buildCommand: Setting;
+    /**
+     * Default dev command or a placeholder
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
     * Default Development Command or a placeholder
```

File: packages/frameworks/src/types.ts [hunk 6]
```diff
@@ -1,50 +1,174 @@
...
   getOutputDirName: (dirPrefix: string) => Promise<string>;
+  /**
+   * An array (or a function that returns an array) of default `Route` rules that
+   * the framework uses.
+   * @example [{ handle: 'filesystem' }, { src: '.*', status: 404, dest: '404.html' }]
+   */
   defaultRoutes?: Route[] | ((dirPrefix: string) => Promise<Route[]>);
+  /**
+   * A glob string of files to cache for future deployments.
+   * @example .nuxt/**
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
   * @example ".cache/**"
```

File: packages/frameworks/src/types.ts [hunk 7]
```diff
@@ -1,50 +1,174 @@
...
+   */
   cachePattern?: string;
+  /**
+   * The default build command for the framework.
+   * @example nuxt generate
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
   * @example next build
```

Let's stick with Next.js for the example to be consistent

File: packages/frameworks/src/types.ts [hunk 8]
```diff
@@ -1,50 +1,174 @@
...
+   */
   buildCommand: string | null;
+  /**
+   * The default development command for the framework.
+   * @example nuxt
```
Comment (styfle) [inline packages/frameworks/src/types.ts]:
```suggestion
   * @example next dev
```

File: packages/frameworks/src/types.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/frameworks/src/types.ts]: I'm gonna leave it for now to be consistent with the others

File: packages/frameworks/src/types.ts [hunk 9]
```diff
@@ -1,50 +1,174 @@
...
+   */
   sort?: number;
-  useRuntime?: { src: string; use: string };
+  /**
+   * Runtime configuration required to run the framework in Vercel
+   */
+  useRuntime?: {
+    /**
+     * Runtime source
+     * @example "package.json"
```
Comment (TooTallNate) [inline packages/frameworks/src/types.ts:84]:
```suggestion
     * @example package.json
```

**Review by styfle — DISMISSED** [review]

**Review by styfle — APPROVED** [review]

**Pull request merged**